hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/EventService.java

```diff
@@ -58,12 +60,11 @@
      * @param serviceName service name
      * @param topic       topic name
      * @param listener    listener instance
-     * @return event registration
+     * @return event registration future
      * @throws IllegalArgumentException if the listener is {@code null}
      */
-    EventRegistration registerLocalListener(@Nonnull String serviceName,
-                                            @Nonnull String topic,
-                                            @Nonnull Object listener);
+    CompletableFuture<EventRegistration> registerLocalListener(@Nonnull String serviceName, @Nonnull String topic,
```

Comment:
i think, it can be better not to introduce unneeded functionality, async local calls are not required as far as i see.